packages/pages-shared/asset-server/handler.ts

```diff
@@ -481,7 +481,19 @@ export async function generateHandler<
 	if (responseWithoutHeaders.status >= 500) {
 		return responseWithoutHeaders;
 	}
-	return await attachHeaders(responseWithoutHeaders);
+
+	const finalResponse = await attachHeaders(responseWithoutHeaders);
```

Comment:
nit: I'd probably call this `responseWithHeaders` as it sounds a bit clearer then `finalResponse`, but that might be just me 😅 :+1: